test/space/fspace_test.py

```diff
@@ -47,49 +46,673 @@ def exponent(request):
+
+
+def _op(a, op, b):
+    if op == '+':
+        return a + b
+    elif op == '-':
+        return a - b
+    elif op == '*':
+        return a * b
+    elif op == '/':
+        return a / b
+    if op == '+=':
+        a += b
+        return a
+    elif op == '-=':
+        a -= b
+        return a
+    elif op == '*=':
+        a *= b
+        return a
+    elif op == '/=':
+        a /= b
+        return a
+
+
+def _test_fspace_vector_op(op_str, pattern):
+    if op_str not in ('+', '-', '*', '/', '+=', '-=', '*=', '/='):
+        raise ValueError('bad operator {!r}.'.format(op_str))
+
+    if pattern not in ('sv', 'vv', 'vs'):
+        raise ValueError('bad pattern {!r}'.format(pattern))
+
+    # Setup
+    rect, points, mg = _standard_setup_2d()
+    point = points.T[0]
+
+    fspace = FunctionSpace(rect)
+    a = -1.5
+    b = 2.0
+    array_out = np.empty((5,), dtype=float)
+    mg_out = np.empty((2, 3), dtype=float)
+
+    # Initialize a bunch of elements
+    f_novec = fspace.element(func_2d_novec, vectorized=False)
+    f_vec = fspace.element(func_2d_vec_dual, vectorized=True)
+    g_novec = fspace.element(other_func_2d_novec, vectorized=False)
+    g_vec = fspace.element(other_func_2d_vec_dual, vectorized=True)
+
+    out_novec = fspace.element(vectorized=False)
+    out_vec = fspace.element(vectorized=True)
+
+    if pattern[0] == 'v':
+        true_l_novec = func_2d_novec(point)
+        true_l_arr = func_2d_vec_oop(points)
+        true_l_mg = func_2d_vec_oop(mg)
+
+        test_l_novec = f_novec
+        test_l_vec = f_vec
+    else:  # 's'
+        true_l_novec = true_l_arr = true_l_mg = a
+        test_l_novec = test_l_vec = a
+
+    if pattern[1] == 'v':
+        true_r_novec = other_func_2d_novec(point)
+        true_r_arr = other_func_2d_vec_oop(points)
+        true_r_mg = other_func_2d_vec_oop(mg)
+
+        test_r_novec = g_novec
+        test_r_vec = g_vec
+    else:  # 's'
+        true_r_novec = true_r_arr = true_r_mg = b
+        test_r_novec = test_r_vec = b
+
+    true_novec = _op(true_l_novec, op_str, true_r_novec)
+    true_arr = _op(true_l_arr, op_str, true_r_arr)
+    true_mg = _op(true_l_mg, op_str, true_r_mg)
+
+    out_novec = _op(test_l_novec, op_str, test_r_novec)
+    out_vec = _op(test_l_vec, op_str, test_r_vec)
+
+    assert out_novec(point) == true_novec
+    assert all_equal(out_vec(points), true_arr)
+    out_vec(points, out=array_out)
+    assert all_equal(array_out, true_arr)
+    assert all_equal(out_vec(mg), true_mg)
+    out_vec(mg, out=mg_out)
+    assert all_equal(mg_out, true_mg)
+
+
+def test_fspace_vector_add():
+    _test_fspace_vector_op('+', 'vv')  # vec + vec
+    _test_fspace_vector_op('+', 'vs')  # vec + scal
+    _test_fspace_vector_op('+', 'sv')  # scal + vec
+    _test_fspace_vector_op('+=', 'vv')  # vec += vec
+    _test_fspace_vector_op('+=', 'vs')  # vec += scal
+
+
+def test_fspace_vector_sub():
+    _test_fspace_vector_op('-', 'vv')  # vec - vec
```

Comment:
Yes, that's very much possible. I think this stuff is from the time before we had fixtures.